src/contracts/TopChef.sol

```diff
@@ -38,23 +37,24 @@ contract TopChef is Chef {
         _allocations.push(group);
         addTotalAllocShares(group.shares);
     }
-    
+
     function updateAllocationGroup(
         address groupAddress,
         uint256 agIndex,
         uint256 shares,
         bool newAutoDistribute
-    ) public onlyOwner() {
+    ) public onlyOwner {
         if (areRewardsActive() && getTotalAllocationShares() > 0) {
             updateAccumulatedAllocations();
         }
         addTotalAllocShares(_allocations[agIndex].shares, shares);
         _allocations[agIndex].groupAddress = groupAddress;
         _allocations[agIndex].shares = shares;
         _allocations[agIndex].autodistribute = newAutoDistribute;
+        _allocations[agIndex].rewardDebt = shares.mul(getLifetimeShareValue()).div(ACC_METRIC_PRECISION).sub(_allocations[agIndex].rewardDebt);
```

Comment:
actually wait - if it's a call to update, they already have existing shares and are earning rewards right?  so would we want to zero it out?

Reply:
when we are updating I thought we were essentially resetting?
Anytime we add an allocation group we do
```newShares.mul(getLifetimeShareValue()).div(ACC_METRIC_PRECISION)```
so I thought we should essentially do the same minus their existing reward debt which would be like, in my view you are just resetting it back to what it would be when you do an additional add - but I might be wrong about this
@addyxvii do you have any thoughts about this?

Reply:
_allocations[agIndex].rewardDebt = shares.mul(getLifetimeShareValue()).div(ACC_METRIC_PRECISION).sub(_allocations[agIndex].rewardDebt);
which does the same thing as harvest I believe without transferring or the claiming of tokens
for example
uint256 claimable = group.shares.mul(getLifetimeShareValue()).div(ACC_METRIC_PRECISION).sub(group.rewardDebt);
group.rewardDebt = claimable;
that statement in harvest (right above) essentially assigns the same value to rewardDebt as the new statement I created in updateAllocationGroup (first code block)
At least that's what I perceive that is why I did it that way